.github/workflows/release.yml

```diff
@@ -48,7 +48,7 @@ jobs:
         run: python -u ./dev/releases/make_archives.py
 
       - name: "Upload artifacts"
-        if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
+        if: ${{ !startsWith(github.ref, 'refs/tags/v') && failure() }}
```

Comment:
nice!

Can we test this? As in, let's check that the current job does not produce an artifact; then perhaps @ssiccha can push a commit which introduces a failure somewhere to see what happens?

Reply:
This is such a niice feature. This did not upload an artifact: 
https://github.com/gap-system/gap/pull/4307/checks?check_run_id=2021086422 
I'll now push something that fails.

Reply:
This works. Here I introduced an error and it is currently uploading artifacts:
https://github.com/gap-system/gap/pull/4307/checks?check_run_id=2021529089

I'm removing my error commit again